src/main/java/org/candlepin/subscriptions/tally/TallySnapshotController.java

```diff
@@ -129,13 +134,16 @@ public void produceHourlySnapshotsForAccount(String accountNumber, DateRange sna
                         .filter(TallySnapshotController::isCombiningRollupStrategySupported)
                         .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
 
-                combiningRollupSnapshotStrategy.produceSnapshotsFromCalculations(
-                    accountNumber,
-                    result.getRange(),
-                    tagProfile.getTagsForServiceType(serviceType),
-                    applicableUsageCalculations,
-                    Granularity.HOURLY,
-                    Double::sum);
+                Map<String, List<TallySnapshot>> totalSnapshots =
+                    combiningRollupSnapshotStrategy.produceSnapshotsFromCalculations(
+                        accountNumber,
+                        result.getRange(),
+                        tagProfile.getTagsForServiceType(serviceType),
+                        applicableUsageCalculations,
+                        Granularity.HOURLY,
+                        Double::sum);
+
+                summaryProducer.produceTallySummaryMessages(totalSnapshots);
```

Comment:
This works because `@Transactional` is on `CombiningRollupSnapshotStrategy#produceSnapshotsFromCalculations`... and thus the DB transaction is committed before the kafka messages are sent. If we ever moved `@Transactional` to `TallySnapshotController#produceHourlySnapshotsForAccount` or called this method from a transaction, it would cause a race again. I'm a little wary of this; can you please add `@Transactional(propagation = Propagation.NEVER)` to this method with a note explaining?